### PR TITLE
:new: Re-design access point selection

### PR DESCRIPTION
**Pull request by blackHatMonkey**
This patch will use curses instead of the default print statements to display the access points. Also all the access point selection has been moved to it's own module called `accesspoint`.

### CODE REVIEW

File: wifiphisher/accesspoint.py [hunk 1]
```diff
@@ -0,0 +1,489 @@
...
+        """
+        Return the name(ESSID) of the access point
+
+        :param self: An AccessPoint object
+        :type self: AccessPoint
+        :return: Name of the access point
+        :rtype: string
+        """
+
+        self.logger.info("Returning name")
```
Comment (sophron) [inline wifiphisher/accesspoint.py]: You are logging way too much information and our logs will lose their usefulness. <a href="https://dzone.com/articles/application-logging-what-when">This guide</a> gives some insight on what and when to log. It's for Java, but i think applies to every language.

Comment (blackHatMonkey) [inline wifiphisher/accesspoint.py]: This is for debugging purposes only. I think we should log anything of warning and higher by default.

**Comment by blackHatMonkey** [comment]
Review the changes and let me know what you think.

File: wifiphisher/pywifiphisher.py [hunk 2]
```diff
@@ -756,25 +1000,27 @@ def run():
...
     print '[' + T + '*' + W + '] Cleared leases, started DHCP, set up iptables'
 
-    if args.essid:
-        essid = args.essid
-        channel = str(CHANNEL)
-        args.accesspoint = False
-        args.channel = False
+    if args.ap_info:
+        essid = args.ap_info[0]
+        channel = args.ap_info[1]
```
Comment (sophron) [inline wifiphisher/pywifiphisher.py]: I think you should remove the `args.ap_info` from here as well.

**Comment by blackHatMonkey** [comment]
Done :smile:.

File: wifiphisher/accesspoint.py (hunk 1 above)
Comment (blackHatMonkey) [inline wifiphisher/accesspoint.py]: Yes, I was thinking of using `logger.debug` option for more in depth logging.

Comment (blackHatMonkey) [inline wifiphisher/accesspoint.py]: perhaps we can leave as is for now and once we design the logging we can modify it.

**Comment by kn0wledge** [comment]
That looks good ! When is the release happening ;)

**Comment by blackHatMonkey** [comment]
@sophron is currently testing it and after that it should be merged.

File: wifiphisher/accesspoint.py (hunk 1 above)
Comment (sophron) [inline wifiphisher/accesspoint.py]: I think our best shot is to remove all logging for now and introduce it separately, after some discussion.

Comment (blackHatMonkey) [inline wifiphisher/accesspoint.py]: sure, I will push a update in a few minutes then.

**Comment by kn0wledge** [comment]
Ok first guy to try the new design 😊. Clean . Number of clients seems to be working , but the power always shows 0% for all APs . I'm not sure if there is color coding depending on the power, if not I would recommend an auto sorting from strongest power to weakest with color coding green yellow and red . Also eventually what you shared , as an improvement to be able to select jamming seperatly so you can jam more than 1 ssid.

**Comment by blackHatMonkey** [comment]
@kn0wledge It should automatically be sorted by power like the following:
modified

If you are not getting this on a fresh install of `Wifiphisher` then open a issue.

**Pull request merged**